fix(group_ldap_link): add missing terraform import feature for group LDAP link

hanks opened the pull request:
**Why**

There is a bug that [the doc](https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs/resources/group_ldap_link#import) says we can use `terraform import`, but actually the function is not implemented

**Code Changes**

* Added the missing import function and acceptance test case
* ~Fixed a bug to set `access_level` instead of `group_access`, because `access_level` is the one defined in schema~
   * ~Assigned the correct string value to `group_access`, before is an int type~, already fixed in the latest main branch
* Set `force=false` as default when importing, because we can not get any value from the Gitlab API
* Updated the doc about import part

**Design Changes**

* Added `group_id` in the import id string to change it from `ldap_provider:cn` to  `group_id:ldap_provider:cn`, because we need the `group_id` to identify the group

**Tests**

* built local provider, and tested
* added acceptance test

**Close**

* #448

timofurrer commented on gitlab/resource_gitlab_group_ldap_link.go:
```diff
@@ -20,6 +20,9 @@ func resourceGitlabGroupLdapLink() *schema.Resource {
 		Read:   resourceGitlabGroupLdapLinkRead,
 		Delete: resourceGitlabGroupLdapLinkDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitlabGroupLdapLinkImporter,
```
It would be nice to use the `schema.ImportStatePassthroughContext` importer instead of writing our custom one.

To make use of that the `Read` function must be able to set the state on the id from `SetId()` only.
See https://www.terraform.io/plugin/sdkv2/resources/import#importer-state-function

I suggest to use the `group_id:lda_provider:cn` three part id. I'm aware that this requires a little refactoring. Please let me know if you want to continue working on this or not.

timofurrer reviewed (changes requested):
Please rebase und run `make reviewable` to generate the documentation.

timofurrer commented:
@hanks thanks for the follow up. Coincidently, I have had a look at this yesterday - see my outcome here #869.

I also doubt that this change isn't backwards-compatible, because you've changed the id. 

Also, the field for the access level is actually called `group_access` and not `access_level` - The `access_level` attribute is deprecated. 

I suggest that we await the outcome of the #869 discussion :)

timofurrer commented:
@hanks if you want to push this further: me and @armsnyder had some discussion in #869 and came to the conclusion that at the moment it might be best to implement a custom importer than can import something like `<group_id>:<ldap_provider>:<cn>` and set the state as it is right now, that is writing the `group` state attribute and id `<ldap_provider>:<cn>`. The reason for this is to not break backwards-compatibility.

Would you like to continue working on this? 🎉

timofurrer reviewed (changes requested):
@hanks if you want to push this further: me and @armsnyder had some discussion in #869 and came to the conclusion that at the moment it might be best to implement a custom importer than can import something like `<group_id>:<ldap_provider>:<cn>` and set the state as it is right now, that is writing the `group` state attribute and id `<ldap_provider>:<cn>`. The reason for this is to not break backwards-compatibility.

Would you like to continue working on this? 🎉

hanks commented:
I tried to add the importer test, and before I added the `SkipFunc` to each step, it failed only in `acceptance-ce` with errors below, and then I added the `SkipFunc`, it seems to pass all the cases. Not sure why in ce test, the `terraform apply` is failed.

```
=== RUN   TestAccGitlabGroupLdapLink_import
    resource_gitlab_group_ldap_link_test.go:73: Step 1/2 error: Error running apply: exit status 1
        
        Error: POST http://127.0.0.1:8080/api/v4/groups/168/ldap_group_links: 404 {error: 404 Not Found}
        
          with gitlab_group_ldap_link.foo,
          on terraform_plugin_test.tf line 8, in resource "gitlab_group_ldap_link" "foo":
           8: resource "gitlab_group_ldap_link" "foo" {
        
--- FAIL: TestAccGitlabGroupLdapLink_import (5.96s)
```

timofurrer commented:
The [LDAP Group Links](https://docs.gitlab.com/ee/api/groups.html#ldap-group-links) are a EE feature :)

timofurrer commented:
Okay, quick update ... There used to be a testdata file to configure the provider, which I've removed a few days ago in #859. 

The original commit https://github.com/gitlabhq/terraform-provider-gitlab/pull/328/files#diff-e864e9b96eaea85189143b708a2f8f57e3c38fe11d57586fd765d6c65274a6f8L29 used to have a warning:

> t.Skipf("[WARNING] Skipping test until test data is configured in %s.", testDataFile)

why the tests are skipped.

Apparently, it was meant to be run with a valid LDAP provider somewhere. However, from this provider perspective we don't actually care too much.
It seems to work fine with using `default` values here.

About the skipped tests I've just created a PR: 

* https://github.com/gitlabhq/terraform-provider-gitlab/pull/873 

I hope that we can quickly merge it and unblock you here - in the meantime you can try and remove the skip func, or at least add the `isRunningInCE` and remove the third step from the `_basic` acceptance test and implement your import test there :)

timofurrer commented:
@hanks In the meantime I've merged #873 which enables the tests - please rebase , it should work for you now :)

hanks commented:
@timofurrer  

> The [LDAP Group Links](https://docs.gitlab.com/ee/api/groups.html#ldap-group-links) are a EE feature

good to know 😄 

thanks for the updates for the test code, and I inserted the import test as the second one, to reuse the Config created from the first step

hanks commented:
@timofurrer Hi, anything else needs to be changed?

timofurrer commented:
@hanks nothing else from your side, just some time from our side! 🎉
